### ISSUE TITLE
Feature request: Add option for auto popup

### ISSUE THREAD

**Issue by mrbiggfoot** [ajh17/VimCompletesMe#53]
I know the goal of the plugin is to be as tiny as possible, but I was wondering how hard it is to add an option to behave like AutoComplPop, i.e. open the completion popup window automatically when a couple of characters is entered?

**Comment by ajh17** [comment]
I’ve had other people also ask me this, and I think it would complicate the plugin, and be slow in general (Autocomplpop has this problem too). I’m not sure if I want to maintain having to fix it if people have issues with it since I wouldn’t use it myself.

That said, I’m willing to take a shot to see if I can do it in a minimal fashion.

**Comment by jaywilliams** [comment]
VimCompletesMe is a fantastic, lightweight plugin. I'd suggest keeping it lightweight and small.

If you're wanting extra features like auto popup, I'd suggest using one of other more feature-full, heavier vim plugins.

**Comment by mrbiggfoot** [comment]
I moved to mucomplete which does what I want. Feel free to close this issue. Thanks!

**Comment by brandonsturgeon** [comment]
I wanted to chime in with a couple of opinions as a new user;
First, I want to make it clear that I absolutely support and appreciate that this project strives for simplicity. I'm okay making some sacrifices to keep it from becoming another YCM.

That being said, I would like this feature, especially because I often work in projects that I didn't write, and it's nice for a few reasons to see that I'm writing a variable/function that's already made or similarly named, etc. Not essential, but useful.

> I’ve had other people also ask me this, and I think it would complicate the plugin, and be slow in general (Autocomplpop has this problem too).

Absolutely makes sense, running on every keypress certainly has some performance implications. Do you have any estimate as to how much it would truly impact performance? Maybe some lazy logic could be done before deciding to pop up the suggestion window.

> If you're wanting extra features like auto popup, I'd suggest using one of other more feature-full, heavier vim plugins.

While I appreciate the sentiment, I think this is a bit of an extreme leap. I understand that to keep something simple means to turn down even small features that may be out of scope, otherwise you start creeping further and further from simplicity. However, this seems like a fairly simple (and common) use case for an autocompletion feature.

Without this feature, it requires the developer to know what can be autocompleted, saving them time only when they know they're re-typing something that already exists (which very well may be the primary use case, don't get me wrong).

As an opt-in feature, this could allow the developer to choose if the performance hit is worth it for them.

> I’m not sure if I want to maintain having to fix it if people have issues with it since I wouldn’t use it myself.
In my opinion this is the most reasonable, understandable, and realistic reason to oppose such a feature. It's not your _job_ to bend to the will of every user (that's what forks are for!), it's very generous of you to support this to the degree you do, and I imagine using it every day makes it a little easier.

So in summary: I would love this feature, I don't know how to write vimscript well enough to implement a good solution to this problem (I'm sure I could throw a hacky inefficient solution), but I completely understand why there would be resistance.

I wanted to throw another perspective in here because it's been about two years since this was last discussed.

Thanks for the tool, it's making my life a lot easier!
